sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ManagedFederatedIdentityCredential.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+
+namespace Microsoft.Extensions.Azure.Internal
+{
+    internal class ManagedFederatedIdentityCredential : TokenCredential
+    {
+        private readonly ManagedIdentityCredential _managedIdentityCredential;
+        private readonly ClientAssertionCredential _clientAssertionCredential;
+
+        /// <summary>
+        ///   Gets the set of additionally allowed tenants.
+        /// </summary>
+        public IList<string> AdditionallyAllowedTenants { get; }
+        /// <summary>
+        /// Creates an instance of the ManagedFederatedIdentityCredential with a synchronous callback that provides a signed client assertion to authenticate against Microsoft Entra ID.
+        /// </summary>
+        /// <param name="tenantId">The Microsoft Entra tenant (directory) ID of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal.</param>
+        /// <param name="managedIdentityClientId">The managed identity which has been configured as a Federated Identity Credential (FIC).</param>
+        /// <param name="federatedAudience">
+        ///     The audience for the federated credential, specific to the cloud.  Valid values are:
+        ///       <list type="bullet">
+        ///         <item>
+        ///           <term>api://AzureADTokenExchange</term>
+        ///           <description>Entra ID Global cloud</description>
+        ///         </item>
+        ///         <item>
+        ///           <term>api://AzureADTokenExchangeUSGov</term>
+        ///           <description>Entra ID US Government</description>
+        ///         </item>
+        ///         <item>
+        ///           <term>api://AzureADTokenExchangeChina</term>
+        ///           <description>Entra ID China operated by 21Vianet</description>
+        ///         </item>
+        ///       </list>
+        /// </param>
+        /// <param name="additionallyAllowedTenants">The set of </param>
+        public ManagedFederatedIdentityCredential(string tenantId, string clientId, string managedIdentityClientId, string federatedAudience, IEnumerable<string> additionallyAllowedTenants = default)
```

Comment:
Since we decided any one of the following can be passed, the `managedIdentityClientId` parameter name should be changed:
- Client ID
- Object ID
- Resource ID

Reply:
Revamped to allow any of the identifiers.